templates/patroni.yml.j2

```diff
@@ -115,6 +115,7 @@ postgresql:
   bin_dir: /usr/lib/postgresql/{{ version }}/bin
   listen: 0.0.0.0:5432
   parameters:
+    shared_preload_libraries: 'timescaledb,pgaudit'
```

Comment:
This was only added to `bootstrap.dcs.postgresql.parameters` (line 48) which AFAIK is only used during bootstrap of the cluster. I think we may be able to remove most of the params in `bootstrap.dcs.postgresql.parameters` but we'll need to investingate what is necessary to be there.

Reply:
@dragomirp please offload DPE-6033 to @marceloneppel . We also need CI test improvement. We have stable=>edge test, I am surprised to see this issue in my manual testing.

Reply:
After testing this PR, I saw that another error was thrown:
```sh
unit-pg-k8s-2: 17:28:32 ERROR unit.pg-k8s/2.juju-log failed to change plugins: 
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pg-k8s-2/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 346, in enable_disable_extensions
    cursor.execute(
psycopg2.errors.UndefinedFile: could not open extension control file "/usr/share/postgresql/14/extension/pgaudit.control": No such file or directory


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-pg-k8s-2/charm/./src/charm.py", line 719, in _handle_enable_disable_extensions
    self.postgresql.enable_disable_extensions(extensions, database)
  File "/var/lib/juju/agents/unit-pg-k8s-2/charm/lib/charms/tempo_coordinator_k8s/v0/charm_tracing.py", line 1064, in wrapped_function
    return callable(*args, **kwargs)  # type: ignore
  File "/var/lib/juju/agents/unit-pg-k8s-2/charm/lib/charms/postgresql_k8s/v0/postgresql.py", line 357, in enable_disable_extensions
    raise PostgreSQLEnableDisableExtensionError()
charms.postgresql_k8s.v0.postgresql.PostgreSQLEnableDisableExtensionError
unit-pg-k8s-2: 17:28:32 INFO unit.pg-k8s/2.juju-log pg-k8s/2 unit upgraded. Evaluate and run `resume-upgrade` action to continue upgrade
```

It happens because the upgraded replica is aware of the new plugins config options, but it tries to enable/disable them in the primary, which is still in the old charm revision.

We can fix it by changing the call to `self.enable_disable_extensions()` in the pebble-ready event handler to:
```python
        if self.upgrade.idle:
            self.enable_disable_extensions()
```

And test with:
```sh
juju deploy postgresql-k8s pg-k8s --trust -n 3 # Please, ensure that the primary is either unit 0 or unit 1.

juju run pg-k8s/leader pre-upgrade-check

juju refresh pg-k8s --channel 14/edge
```

Reply:
@dragomirp shared in MM: this conversation is no longer relevant. The issue for failing test was a wrong OCI resource (addressed here as well).